third_party/move/move-core/types/src/value.rs

```diff
@@ -263,6 +266,10 @@ pub enum MoveTypeLayout {
     // TODO[agg_v2](?): Do we need a layout here if we have custom serde
     //                  implementations available?
     Native(IdentifierMappingKind, Box<MoveTypeLayout>),
+
+    // Added in bytecode version v8
+    #[serde(rename(serialize = "fun", deserialize = "fun"))]
+    Function(MoveFunctionLayout),
```

Comment:
This is rather bad for performance: nested struct that uses a nested function that uses multiple allocations. If we cache layouts (see Slack discussion) - this has very bad performance and we should really use pointers for layouts we intend to cache (e.g., structs). Probably not related to this PR straight away, but if we ever want to cache function layouts (which can make sense, they are resolved by name) - this will bite us.

Reply:
Perhaps you can refactor it once it lands and before it goes to production? But this is what we logical need. 

Note all the fancy caches aren't available in this part of the code, so there need to be a standallone representation of type layout. The size of those layouts is already vetted when they are created at runtime.

The slack discussion is only about layouts in the runtime. I already noted this on the design doc.